content/en/monitors/guide/division-in-monitor-evaluations.md

```diff
@@ -0,0 +1,211 @@
+---
+title: Division in Monitor Evaluations
+kind: guide
+aliases:
+  - /monitors/faq/Division-in-Monitor-Evaluations
+---
+
+## Overview
+
+Creating an alert based on a query with division is a common practice. There are some tools and behaviors that should be considered to ensure monitor settings are appropriate for evaluating these queries as intended.
+
+## Error Rate Example
+
+Suppose you want to monitor an error rate over 5 minutes using the metrics, `requests.error` and `requests.total`. There are 2 possible ways to perform this calculation.
+
+Consider a single evaluation performed with these aligned timeseries points for the 5 min timeframe:
+
+**Numerator**: `sum:requests.error{*}`
+
+```
+| Timestamp             | Value       |
+| :-------------------- | :---------- |
+| 2018-03-13 11:00:30   | 1           |
+| 2018-03-13 11:01:30   | 2           |
+| 2018-03-13 11:02:40   | 3           |
+| 2018-03-13 11:03:30   | 4           |
+| 2018-03-13 11:04:40   | 5           |
+```
+
+**Denominator**: `sum:requests.total{*}`
+
+```
+| Timestamp             | Value    |
+| :-------------------- | :------- |
+| 2018-03-13 11:00:30   | 5        |
+| 2018-03-13 11:01:30   | 5        |
+| 2018-03-13 11:02:40   | 5        |
+| 2018-03-13 11:03:30   | 5        |
+| 2018-03-13 11:04:40   | 5        |
+```
+
+### Division with `as_count()`
+
+Monitors involving division and at least 1 `as_count()` modifier use a separate evaluation path than other monitors. This impacts queries of the format `queryA.as_count() / queryB.as_count()`. Queries without division (`+`, `-`, or `*`) are not impacted.
+
+Refer this query **`as_count_eval_path`**:
+
+```
+sum(last_5m): sum:requests.error{*}.as_count() / sum:requests.total{*}.as_count()
+```
+
+and this query **`classic_eval_path`**\*:
+
+```
+sum(last_5m): sum:requests.error{*}.as_rate() / sum:requests.total{*}.as_rate()
+```
+
+Compare the result of the evaluation depending of the path:
+
+| Path                     | Behavior                                       | Expanded expression         | Result  |
+| :----------------------- | :--------------------------------------------- | :-------------------------- | :------ |
+| **`as_count_eval_path`** | Aggregation function applied _before_ division | **(1+2+...+5)/(5+5+...+5)** | **0.6** |
+| **`classic_eval_path`**  | Aggregation function applied _after_ division  | **(1/5 + 2/5 + ... + 5/5)** | **3**   |
+
+You could also use `avg` and `as_rate()` to get the same result as `sum` and `as_count()` :
```

Comment:
The result would be similar but it wouldn't be the same in general. Consider the following two time series:

```
a = [1, 2, 3, 4, 5]
b = [5, 4, 3, 2, 1]
```

The sum of both series is the same, so the quotient of the sums is 1, but average of the quotients isn't quite 1:

```
const _ = require('lodash');
_.mean([1/5, 2/4, 3/3, 4/2, 5/1])
1.74
```

Average time aggregation with `as_rate()` is reasonable, but it sum/`as_count()` is better for error rates.